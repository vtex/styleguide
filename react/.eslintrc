{
  "extends": "vtex-react",
  "env": {
    "browser": true,
    "es6": true,
    "jest": true
  }
}
