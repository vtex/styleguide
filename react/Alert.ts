import Alert from './components/Alert/index'

export default Alert
