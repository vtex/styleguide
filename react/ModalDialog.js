import ModalDialog from './components/ModalDialog/index'

export default ModalDialog
