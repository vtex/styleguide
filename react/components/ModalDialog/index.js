import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Modal from '../Modal'
import Button from '../Button'

class ModalDialog extends Component {
  handleConfirmation = () => {
    this.props.confirmation &&
      this.props.confirmation.onClick &&
      this.props.confirmation.onClick()
  }

  handleCancelation = () => {
    this.props.cancelation &&
      this.props.cancelation.onClick &&
      this.props.cancelation.onClick()
  }

  render() {
    const { confirmation, cancelation } = this.props

    return (
      <Modal {...this.props}>
        {this.props.children}

        <div className="vtex-modal__confirmation flex justify-end mt8">
          <span className="mr4">
            <Button
              size="small"
              variation="tertiary"
              onClick={this.handleCancelation}
            >
              {cancelation.label}
            </Button>
          </span>

          <Button
            size="small"
            variation="primary"
            onClick={this.handleConfirmation}
          >
            {confirmation.label}
          </Button>
        </div>
      </Modal>
    )
  }
}

ModalDialog.propTypes = {
  children: PropTypes.node,
  confirmation: PropTypes.shape({
    label: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
  }).isRequired,
  cancelation: PropTypes.shape({
    label: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
  }).isRequired,
}

export default ModalDialog
