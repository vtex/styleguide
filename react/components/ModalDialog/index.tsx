import React, { FC } from 'react'

import Modal, { Props as ModalProps } from '../Modal'
import Button from '../Button'

type DialogOption = {
  label: string
  onClick: () => unknown
}

type Props = Pick<
  ModalProps,
  'children' | 'isOpen' | 'onClose' | 'onCloseTransitionFinish'
> & {
  loading?: boolean
  confirmation: DialogOption & { isDangerous?: boolean }
  cancelation: DialogOption
}

const ModalDialog: FC<Props> = ({
  children,
  confirmation,
  cancelation,
  loading,
  ...props
}) => {
  const handleCancelation = () => {
    cancelation.onClick?.()
  }

  const handleConfirmation = () => {
    confirmation.onClick?.()
  }

  return (
    <Modal
      {...props}
      bottomBar={
        <div>
          <Button
            size="small"
            type="button"
            variation="tertiary"
            disabled={loading}
            onClick={handleCancelation}
          >
            {cancelation.label}
          </Button>
          <Button
            size="small"
            type="button"
            variation={confirmation.isDangerous ? 'danger' : 'primary'}
            isLoading={loading}
            onClick={handleConfirmation}
          >
            {confirmation.label}
          </Button>
        </div>
      }
    >
      {children}
    </Modal>
  )
}

export default ModalDialog
