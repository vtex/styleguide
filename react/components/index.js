export { default as ActionMenu } from './ActionMenu'
export { default as Alert } from './Alert'
export { default as AutocompleteInput } from './AutocompleteInput'
export { default as BarChart } from './EXPERIMENTAL_Charts/BarChart'
export { default as Box } from './Box'
export { default as BubbleChart } from './EXPERIMENTAL_Charts/BubbleChart'
export { default as Button } from './Button'
export { default as ButtonGroup } from './ButtonGroup'
export { default as ButtonPlain } from './ButtonPlain'
export { default as ButtonWithIcon } from './ButtonWithIcon'
export { default as Card } from './Card'
export { default as Checkbox } from './Checkbox'
export { default as CheckboxGroup } from './CheckboxGroup'
export { default as Collapsible } from './Collapsible'
export { default as ColorPicker } from './ColorPicker'
export { default as Conditions } from './Conditions'
export { default as DatePicker } from './DatePicker'
export { default as Divider } from './Divider'
export { default as Dropdown } from './Dropdown'
export { default as Dropzone } from './Dropzone'
export { default as EmptyState } from './EmptyState'
export { default as EXPERIMENTAL_Conditions } from './EXPERIMENTAL_Conditions'
export { default as EXPERIMENTAL_Select } from './EXPERIMENTAL_Select'
export { default as EXPERIMENTAL_Table } from './EXPERIMENTAL_Table'
export { default as EXPERIMENTAL_useCheckboxTree } from './EXPERIMENTAL_useCheckboxTree'
export { default as EXPERIMENTAL_useTableMeasures } from './EXPERIMENTAL_Table/hooks/useTableMeasures'
export { default as EXPERIMENTAL_useTableProportion } from './EXPERIMENTAL_Table/hooks/useTableProportion'
export { default as EXPERIMENTAL_useTableSort } from './EXPERIMENTAL_Table/hooks/useTableSort'
export { default as EXPERIMENTAL_useTableVisibility } from './EXPERIMENTAL_Table/hooks/useTableVisibility'
export { default as FilterBar } from './FilterBar'
export { default as FilterOptions } from './FilterOptions'
export { default as FloatingActionBar } from './FloatingActionBar'
export { default as IconArrowBack } from './icon/ArrowBack'
export { default as IconArrowDown } from './icon/ArrowDown'
export { default as IconArrowUp } from './icon/ArrowUp'
export { default as IconBars } from './icon/Bars'
export { default as IconBold } from './icon/Bold'
export { default as IconCalendar } from './icon/Calendar'
export { default as IconCaretDown } from './icon/CaretDown'
export { default as IconCaretLeft } from './icon/CaretLeft'
export { default as IconCaretRight } from './icon/CaretRight'
export { default as IconCaretUp } from './icon/CaretUp'
export { default as IconCheck } from './icon/Check'
export { default as IconCheckPartial } from './icon/CheckPartial' // not a render entrypoint
export { default as IconClear } from './icon/Clear'
export { default as IconClock } from './icon/Clock'
export { default as IconClose } from './icon/Close'
export { default as IconCog } from './icon/Cog'
export { default as IconColumns } from './icon/Columns'
export { default as IconCopy } from './icon/Copy'
export { default as IconDelete } from './icon/Delete'
export { default as IconDensity } from './icon/Density'
export { default as IconDeny } from './icon/Deny'
export { default as IconDownload } from './icon/Download'
export { default as IconEdit } from './icon/Edit'
export { default as IconExternalLink } from './icon/ExternalLink'
export { default as IconExternalLinkMini } from './icon/ExternalLinkMini'
export { default as IconFailure } from './icon/Failure'
export { default as IconFile } from './icon/File'
export { default as IconFilter } from './icon/Filter'
export { default as IconGrid } from './icon/Grid'
export { default as IconHelp } from './icon/Help'
export { default as IconImage } from './icon/Image'
export { default as IconInfo } from './icon/Info'
export { default as IconInlineGrid } from './icon/InlineGrid'
export { default as IconItalic } from './icon/Italic'
export { default as IconLink } from './icon/Link'
export { default as IconLocation } from './icon/Location'
export { default as IconMinus } from './icon/Minus'
export { default as IconOptionsDots } from './icon/OptionsDots'
export { default as IconOrderedList } from './icon/OrderedList'
export { default as IconPlus } from './icon/Plus'
export { default as IconPlusLines } from './icon/PlusLines'
export { default as IconPrinter } from './icon/Printer'
export { default as IconSave } from './icon/Save'
export { default as IconSearch } from './icon/Search'
export { default as IconShoppingCart } from './icon/ShoppingCart'
export { default as IconSuccess } from './icon/Success'
export { default as IconTableFilter } from './icon/TableFilter'
export { default as IconText } from './icon/Text'
export { default as IconUnderline } from './icon/Underline'
export { default as IconUnorderedList } from './icon/UnorderedList'
export { default as IconUpload } from './icon/Upload'
export { default as IconUser } from './icon/User'
export { default as IconVisibilityOff } from './icon/VisibilityOff'
export { default as IconVisibilityOn } from './icon/VisibilityOn'
export { default as IconWarning } from './icon/Warning'
export { default as Input } from './Input'
export { default as InputButton } from './InputButton'
export { default as InputCurrency } from './InputCurrency'
export { default as InputPassword } from './InputPassword'
export { default as InputSearch } from './InputSearch'
export { default as Layout } from './Layout'
export { default as LineChart } from './EXPERIMENTAL_Charts/LineChart'
export { default as Link } from './Link'
export { default as Menu } from './Menu'
export { default as Modal } from './Modal'
export { default as ModalDialog } from './ModalDialog'
export { default as MultiSelect } from './MultiSelect'
export { default as NumericStepper } from './NumericStepper'
export { default as PageBlock } from './PageBlock'
export { default as PageHeader } from './PageHeader'
export { default as Pagination } from './Pagination'
export { default as PasswordInput } from './PasswordInput'
export { default as Progress } from './Progress'
export { default as ProgressBar } from './ProgressBar'
export { default as Radio } from './Radio'
export { default as RadioGroup } from './RadioGroup'
export { default as ScatterChart } from './EXPERIMENTAL_Charts/ScatterChart'
export { default as SelectableCard } from './SelectableCard'
export { default as Slider } from './Slider'
export { default as Spinner } from './Spinner'
export { default as Tab } from './Tab'
export { default as Table } from './Table'
export { default as Tabs } from './Tabs'
export { default as Tag } from './Tag'
export { default as Textarea } from './Textarea'
export { default as TimePicker } from './TimePicker'
export { default as Toggle } from './Toggle'
export { default as Tooltip } from './Tooltip'
export { default as Totalizer } from './Totalizer'
export {
  ToastContext,
  ToastConsumer,
  ToastProvider,
  withToast,
} from './ToastProvider'
