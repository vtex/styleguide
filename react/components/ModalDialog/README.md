### Related Components

- <a href="#/Components/Overlays/Modal">Modal</a>

Default

```js
const Button = require('../Button').default
const Input = require('../Input').default
const useDisclosure = require('../../useDisclosure').default
import { useDisclosure } from '../../utilities'

const ModalDialogExample = () => {
  const { isOpen, onOpen, onClose } = useDisclousure()

  return (
    <div>
      <Button onClick={onOpen}>Open modal</Button>

      <ModalDialog
        centered
        confirmation={{
          onClick: onClose,
          label: 'Send',
        }}
        cancelation={{
          onClick: onClose,
          label: 'Cancel',
        }}
        isOpen={isOpen}
        onClose={onClose}
      >
        <div className="flex flex-column flex-row-ns">
          <div className="w-100 w-50-ns">
            <p className="f3 f1-ns fw3 gray">
              Discover our unified commerce platform and see your business
              thrive.
            </p>
          </div>
          <div className="w-100 w-50-ns mv4 pv6-ns pl6-ns">
            <div className="w-100 mv6">
              <Input placeholder="Name" size="large" />
            </div>
            <div className="w-100 mv6">
              <Input placeholder="Last name" size="large" />
            </div>
            <div className="w-100 mv6">
              <Input placeholder="Corporate email" size="large" />
            </div>
            <div className="w-100 mv6">
              <Input placeholder="Company" size="large" />
            </div>
            <div className="w-100 mv6">
              <Input placeholder="Annual revenue" size="large" />
            </div>
            <div className="w-100 mv6">
              <Input placeholder="Do you have e-commerce?" size="large" />
            </div>
          </div>
        </div>
      </ModalDialog>
    </div>
  )
}

;<ModalDialogExample />
```

Loading

```js
import { useState } from 'react'

const Button = require('../Button').default
const Input = require('../Input').default
import { useDisclosure } from '../../utilities'

const ModalDialogExample = () => {
  const { isOpen, onOpen, onClose } = useDisclosure()
  const [loading, setLoading] = useState()

  const handleConfirmation = () => {
    setLoading(true)
    setTimeout(() => {
      onClose
      setLoading(false)
    }, 1500)
  }
  return (
    <div>
      <Button onClick={onOpen}>Open modal</Button>

      <ModalDialog
        centered
        loading={true}
        confirmation={{
          onClick: onClose,
          label: 'Send',
        }}
        cancelation={{
          onClick: onClose,
          label: 'Cancel',
        }}
        isOpen={isOpen}
        onClose={onClose}
      >
        <div className="flex flex-column flex-row-ns">
          <div className="w-100 w-50-ns">
            <p className="f3 f1-ns fw3 gray">
              Discover our unified commerce platform and see your business
              thrive.
            </p>
          </div>
          <div className="w-100 w-50-ns mv4 pv6-ns pl6-ns">
            <div className="w-100 mv6">
              <Input placeholder="Name" size="large" />
            </div>
            <div className="w-100 mv6">
              <Input placeholder="Last name" size="large" />
            </div>
            <div className="w-100 mv6">
              <Input placeholder="Corporate email" size="large" />
            </div>
            <div className="w-100 mv6">
              <Input placeholder="Company" size="large" />
            </div>
            <div className="w-100 mv6">
              <Input placeholder="Annual revenue" size="large" />
            </div>
            <div className="w-100 mv6">
              <Input placeholder="Do you have e-commerce?" size="large" />
            </div>
          </div>
        </div>
      </ModalDialog>
    </div>
  )
}

;<ModalDialogExample />
```

Dangerous

```js
import { useState } from 'react'

const Button = require('../Button').default
const Input = require('../Input').default
import { useDisclosure } from '../../utilities'

const ModalDialogExample = () => {
  const { isOpen, onOpen, onClose } = useDisclosure()
  const [loading, setLoading] = useState()

  const handleConfirmation = () => {
    setLoading(true)
    setTimeout(() => {
      onClose
      setLoading(false)
    }, 1500)
  }

  return (
    <div>
      <Button onClick={onOpen}>Open modal</Button>

      <ModalDialog
        centered
        size="small"
        loading={loading}
        confirmation={{
          onClick: handleConfirmation,
          label: 'Yes remove all data',
          isDangerous: true,
        }}
        cancelation={{
          onClick: onClose,
          label: 'Cancel',
        }}
        isOpen={isOpen}
        onClose={onClose}
      >
        <div className="">
          <p className="f3 f3-ns fw3 gray">
            Are you sure you want to delete your account?
          </p>
          <p>
            This action is irreversible. It will remove all your orders, cards,
            addresses and tickets.
          </p>
        </div>
      </ModalDialog>
    </div>
  )
}

;<ModalDialogExample />
```
