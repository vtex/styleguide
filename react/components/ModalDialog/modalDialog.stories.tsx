import React from 'react'
import { withA11y } from '@storybook/addon-a11y'

import Button from '../Button'
import ModalDialog from '.'
import useModal from '../Modal/useModal'

export default {
  title: 'Components|ModalDialog',
  decorators: [withA11y],
}

export const Default = () => {
  const { isOpen, open, close } = useModal()

  return (
    <>
      <Button onClick={open} type="button">
        Open
      </Button>
      <ModalDialog
        isOpen={isOpen}
        onClose={close}
        confirmation={{
          label: 'Confirm',
          onClick: close,
          isDangerous: true,
        }}
        cancelation={{
          label: 'Cancel',
          onClick: close,
        }}
      >
        Content
      </ModalDialog>
    </>
  )
}
