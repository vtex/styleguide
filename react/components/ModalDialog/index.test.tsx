import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import ModalDialog from '.'

describe('ModalDialog', () => {
  jest.spyOn(window, 'scroll').mockImplementation()
  it('should have a default export', () => {
    expect(typeof ModalDialog).toBe('function')
  })

  describe('cancelation', () => {
    it('should render label', () => {
      const containerModal = document.createElement('div')
      const onCancel = jest.fn()
      const cancelLabel = 'cancel'
      const { getByText, container } = render(
        <ModalDialog
          isOpen
          container={containerModal}
          cancelation={{
            label: cancelLabel,
            onClick: onCancel,
          }}
          confirmation={{
            label: 'confirm',
            onClick: () => null,
          }}
        >
          Foo
        </ModalDialog>,
        { container: containerModal }
      )
      expect(getByText(cancelLabel)).not.toBeNull()
      expect(container).toMatchSnapshot()
    })

    it('onClick should be called on cancelation button click', () => {
      const onCancel = jest.fn()

      const { getByText } = render(
        <ModalDialog
          isOpen
          cancelation={{
            label: 'cancel',
            onClick: onCancel,
          }}
          confirmation={{
            label: 'confirm',
            onClick: () => null,
          }}
        >
          Foo
        </ModalDialog>,
        { container: document.body }
      )

      fireEvent.click(getByText('cancel'))
      expect(onCancel).toHaveBeenCalled()
    })
  })

  describe('confirmation', () => {
    it('should render label', () => {
      const containerModal = document.createElement('div')
      const onConfirm = jest.fn()
      const confirmLabel = 'confirm'
      const { getAllByText, container } = render(
        <ModalDialog
          isOpen
          container={containerModal}
          cancelation={{
            label: 'cancel',
            onClick: () => null,
          }}
          confirmation={{
            label: confirmLabel,
            onClick: onConfirm,
          }}
        >
          Foo
        </ModalDialog>,
        { container: containerModal }
      )
      expect(getAllByText(confirmLabel)).not.toBeNull()
      expect(container).toMatchSnapshot()
    })

    it('onClick should be called on confirmation button click', () => {
      const onConfirm = jest.fn()
      const confirmLabel = 'confirm'
      const { getByText } = render(
        <ModalDialog
          isOpen
          cancelation={{
            label: 'cancel',
            onClick: () => null,
          }}
          confirmation={{
            label: confirmLabel,
            onClick: onConfirm,
          }}
        >
          Foo
        </ModalDialog>,
        { container: document.body }
      )

      fireEvent.click(getByText(confirmLabel))
      expect(onConfirm).toHaveBeenCalled()
    })

    it('isDangerous CSS', () => {
      const containerModal = document.createElement('div')
      const onConfirm = jest.fn()
      const confirmLabel = 'cancel'
      const { container } = render(
        <ModalDialog
          isOpen
          container={containerModal}
          cancelation={{
            label: 'cancel',
            onClick: () => null,
          }}
          confirmation={{
            label: confirmLabel,
            onClick: onConfirm,
            isDangerous: true,
          }}
        >
          Foo
        </ModalDialog>,
        { container: containerModal }
      )

      expect(container).toMatchSnapshot()
    })
  })

  describe('loading', () => {
    it('CSS', () => {
      const containerModal = document.createElement('div')
      const { container } = render(
        <ModalDialog
          isOpen
          loading
          container={containerModal}
          cancelation={{
            label: 'cancel',
            onClick: () => null,
          }}
          confirmation={{
            label: 'confirm',
            onClick: () => null,
            isDangerous: true,
          }}
        >
          Foo
        </ModalDialog>,
        { container: containerModal }
      )

      expect(container).toMatchSnapshot()
    })
  })
})
