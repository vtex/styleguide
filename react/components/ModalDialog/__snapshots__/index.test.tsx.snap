// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalDialog cancelation should render label 1`] = `
<div>
  <div
    class="flex fixed z-max overflow-hidden overflow-hidden bg-black-70 absolute--fill items-center"
    data-testid="modal__overlay"
    tabindex="-1"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      class="undefined"
      data-focus-lock-disabled="false"
    >
      <div
        aria-labelledby="vtex-modal__title"
        aria-modal="true"
        class="flex flex-column relative bg-white shadow-5 center mv9 vw-50-ns vw-50"
        data-testid="modal__modal"
        role="dialog"
      >
        <div
          class="mb5"
        >
          <div
            class="min-h-large pl8"
          >
            <div
              class="fr pointer pr6-ns pt6-ns pr5 pt5"
              role="button"
              tabindex="0"
            >
              <svg
                class="vtex__icon-close  "
                fill="none"
                height="16"
                viewBox="0 0 18 18"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  fill="black"
                >
                  <path
                    d="M14.53 4.53l-1.06-1.06L9 7.94 4.53 3.47 3.47 4.53 7.94 9l-4.47 4.47 1.06 1.06L9 10.06l4.47 4.47 1.06-1.06L10.06 9z"
                  />
                </g>
              </svg>
            </div>
          </div>
          <span
            class="t-heading-4 ml8"
          >
            <div
              class="undefined"
              id="vtex-modal__title"
            />
          </span>
        </div>
        <div
          class="ph8 t-body overflow-auto flex flex-column flex-shrink-1 flex-grow-1 mb3 undefined undefined"
        >
          Foo
        </div>
        <div
          class="flex justify-content flex-row-reverse min-h-regular-ns min-h-small pv6-ns ph8-ns pv7 ph6 mt3 bt b--muted-4"
        >
          <div>
            <div>
              <button
                class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary pointer "
                tabindex="0"
                type="button"
              >
                <div
                  class="vtex-button__label flex items-center justify-center h-100 ph5 "
                  style="padding-top: .25em; padding-bottom: .32em;"
                >
                  cancel
                </div>
              </button>
              <button
                class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-action-primary b--action-primary c-on-action-primary hover-bg-action-primary hover-b--action-primary hover-c-on-action-primary pointer "
                tabindex="0"
                type="button"
              >
                <div
                  class="vtex-button__label flex items-center justify-center h-100 ph5 "
                  style="padding-top: .25em; padding-bottom: .32em;"
                >
                  confirm
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`ModalDialog confirmation isDangerous CSS 1`] = `
<div>
  <div
    class="flex fixed z-max overflow-hidden overflow-hidden bg-black-70 absolute--fill items-center"
    data-testid="modal__overlay"
    tabindex="-1"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      class="undefined"
      data-focus-lock-disabled="false"
    >
      <div
        aria-labelledby="vtex-modal__title"
        aria-modal="true"
        class="flex flex-column relative bg-white shadow-5 center mv9 vw-50-ns vw-50"
        data-testid="modal__modal"
        role="dialog"
      >
        <div
          class="mb5"
        >
          <div
            class="min-h-large pl8"
          >
            <div
              class="fr pointer pr6-ns pt6-ns pr5 pt5"
              role="button"
              tabindex="0"
            >
              <svg
                class="vtex__icon-close  "
                fill="none"
                height="16"
                viewBox="0 0 18 18"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  fill="black"
                >
                  <path
                    d="M14.53 4.53l-1.06-1.06L9 7.94 4.53 3.47 3.47 4.53 7.94 9l-4.47 4.47 1.06 1.06L9 10.06l4.47 4.47 1.06-1.06L10.06 9z"
                  />
                </g>
              </svg>
            </div>
          </div>
          <span
            class="t-heading-4 ml8"
          >
            <div
              class="undefined"
              id="vtex-modal__title"
            />
          </span>
        </div>
        <div
          class="ph8 t-body overflow-auto flex flex-column flex-shrink-1 flex-grow-1 mb3 undefined undefined"
        >
          Foo
        </div>
        <div
          class="flex justify-content flex-row-reverse min-h-regular-ns min-h-small pv6-ns ph8-ns pv7 ph6 mt3 bt b--muted-4"
        >
          <div>
            <div>
              <button
                class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary pointer "
                tabindex="0"
                type="button"
              >
                <div
                  class="vtex-button__label flex items-center justify-center h-100 ph5 "
                  style="padding-top: .25em; padding-bottom: .32em;"
                >
                  cancel
                </div>
              </button>
              <button
                class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-danger b--danger c-on-danger hover-bg-danger hover-b--danger hover-c-on-danger pointer "
                tabindex="0"
                type="button"
              >
                <div
                  class="vtex-button__label flex items-center justify-center h-100 ph5 "
                  style="padding-top: .25em; padding-bottom: .32em;"
                >
                  cancel
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`ModalDialog confirmation should render label 1`] = `
<div>
  <div
    class="flex fixed z-max overflow-hidden overflow-hidden bg-black-70 absolute--fill items-center"
    data-testid="modal__overlay"
    tabindex="-1"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      class="undefined"
      data-focus-lock-disabled="false"
    >
      <div
        aria-labelledby="vtex-modal__title"
        aria-modal="true"
        class="flex flex-column relative bg-white shadow-5 center mv9 vw-50-ns vw-50"
        data-testid="modal__modal"
        role="dialog"
      >
        <div
          class="mb5"
        >
          <div
            class="min-h-large pl8"
          >
            <div
              class="fr pointer pr6-ns pt6-ns pr5 pt5"
              role="button"
              tabindex="0"
            >
              <svg
                class="vtex__icon-close  "
                fill="none"
                height="16"
                viewBox="0 0 18 18"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  fill="black"
                >
                  <path
                    d="M14.53 4.53l-1.06-1.06L9 7.94 4.53 3.47 3.47 4.53 7.94 9l-4.47 4.47 1.06 1.06L9 10.06l4.47 4.47 1.06-1.06L10.06 9z"
                  />
                </g>
              </svg>
            </div>
          </div>
          <span
            class="t-heading-4 ml8"
          >
            <div
              class="undefined"
              id="vtex-modal__title"
            />
          </span>
        </div>
        <div
          class="ph8 t-body overflow-auto flex flex-column flex-shrink-1 flex-grow-1 mb3 undefined undefined"
        >
          Foo
        </div>
        <div
          class="flex justify-content flex-row-reverse min-h-regular-ns min-h-small pv6-ns ph8-ns pv7 ph6 mt3 bt b--muted-4"
        >
          <div>
            <div>
              <button
                class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-transparent b--transparent c-action-primary hover-b--transparent hover-bg-action-secondary hover-b--action-secondary pointer "
                tabindex="0"
                type="button"
              >
                <div
                  class="vtex-button__label flex items-center justify-center h-100 ph5 "
                  style="padding-top: .25em; padding-bottom: .32em;"
                >
                  cancel
                </div>
              </button>
              <button
                class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-action-primary b--action-primary c-on-action-primary hover-bg-action-primary hover-b--action-primary hover-c-on-action-primary pointer "
                tabindex="0"
                type="button"
              >
                <div
                  class="vtex-button__label flex items-center justify-center h-100 ph5 "
                  style="padding-top: .25em; padding-bottom: .32em;"
                >
                  confirm
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`ModalDialog loading CSS 1`] = `
<div>
  <div
    class="flex fixed z-max overflow-hidden overflow-hidden bg-black-70 absolute--fill items-center"
    data-testid="modal__overlay"
    tabindex="-1"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      class="undefined"
      data-focus-lock-disabled="false"
    >
      <div
        aria-labelledby="vtex-modal__title"
        aria-modal="true"
        class="flex flex-column relative bg-white shadow-5 center mv9 vw-50-ns vw-50"
        data-testid="modal__modal"
        role="dialog"
      >
        <div
          class="mb5"
        >
          <div
            class="min-h-large pl8"
          >
            <div
              class="fr pointer pr6-ns pt6-ns pr5 pt5"
              role="button"
              tabindex="0"
            >
              <svg
                class="vtex__icon-close  "
                fill="none"
                height="16"
                viewBox="0 0 18 18"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
                xmlns:xlink="http://www.w3.org/1999/xlink"
              >
                <g
                  fill="black"
                >
                  <path
                    d="M14.53 4.53l-1.06-1.06L9 7.94 4.53 3.47 3.47 4.53 7.94 9l-4.47 4.47 1.06 1.06L9 10.06l4.47 4.47 1.06-1.06L10.06 9z"
                  />
                </g>
              </svg>
            </div>
          </div>
          <span
            class="t-heading-4 ml8"
          >
            <div
              class="undefined"
              id="vtex-modal__title"
            />
          </span>
        </div>
        <div
          class="ph8 t-body overflow-auto flex flex-column flex-shrink-1 flex-grow-1 mb3 undefined undefined"
        >
          Foo
        </div>
        <div
          class="flex justify-content flex-row-reverse min-h-regular-ns min-h-small pv6-ns ph8-ns pv7 ph6 mt3 bt b--muted-4"
        >
          <div>
            <div>
              <button
                class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-transparent b--transparent c-disabled "
                disabled=""
                tabindex="0"
                type="button"
              >
                <div
                  class="vtex-button__label flex items-center justify-center h-100 ph5 "
                  style="padding-top: .25em; padding-bottom: .32em;"
                >
                  cancel
                </div>
              </button>
              <button
                class="vtex-button bw1 ba fw5 v-mid relative pa0 lh-solid br2 min-h-small t-action--small bg-disabled b--muted-5 c-on-disabled "
                tabindex="0"
                type="button"
              >
                <span
                  class="vtex-button__spinner-container top-0 left-0 w-100 h-100 absolute flex justify-center items-center"
                >
                  <svg
                    class="vtex__icon-spinner  c-action-primary"
                    height="15"
                    preserveAspectRatio="xMidYMid"
                    viewBox="0 0 100 100"
                    width="15"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <style>
                      
            @keyframes vtex-spinner-rotate {
              from {
                transform-origin: 50% 50%;
                transform: rotate(0deg);
              }
              to {
                transform-origin: 50% 50%;
                transform: rotate(360deg);
              }
            }

            @keyframes vtex-spinner-fill {
              0% {
                stroke-dasharray: 0 0 2 251.32741228718345;
              }
              50% {
                stroke-dasharray: 0 0 188.49555921538757 251.32741228718345;
              }
              100% {
                stroke-dasharray: 0 249.32741228718345 188.49555921538757 251.32741228718345;
              }
            }

            .vtex-spinner_circle {
              animation: vtex-spinner-fill 1.25s infinite cubic-bezier(0.455, 0.030, 0.515, 0.955), vtex-spinner-rotate 0.625s infinite linear;
            }
          
                    </style>
                    <circle
                      class="vtex-spinner_circle"
                      cx="50"
                      cy="50"
                      fill="none"
                      r="40"
                      stroke="currentColor"
                      stroke-dasharray="0 0 188.49555921538757 251.32741228718345"
                      stroke-dashoffset="1"
                      stroke-linecap="round"
                      stroke-width="10"
                    />
                  </svg>
                </span>
                <span
                  class="vtex-button__label flex items-center justify-center h-100 ph5  o-0"
                >
                  confirm
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</div>
`;
